File: src/test/java/com/lakamsani/farefinder/AirportTest.java
package com.lakamsani.farefinder;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class AirportTest {

    @Test
    public void testKnownAirports() throws Exception {
        assertEquals("SFO", Airport.lookupAirport("San Francisco", "CA"));
        assertEquals("JFK", Airport.lookupAirport("New York", "NY"));
        assertEquals("ORD", Airport.lookupAirport("Chicago", "IL"));
        assertEquals("ATL", Airport.lookupAirport("Atlanta", "GA"));
        assertEquals("DEN", Airport.lookupAirport("Denver", "CO"));
        assertEquals("SEA", Airport.lookupAirport("Seattle", "WA"));
        assertEquals("MIA", Airport.lookupAirport("Miami", "FL"));
        assertEquals("LAS", Airport.lookupAirport("Las Vegas", "NV"));
        assertEquals("BOS", Airport.lookupAirport("Boston", "MA"));
        assertEquals("DFW", Airport.lookupAirport("Dallas", "TX"));
    }

    @Test
    public void testUnknownCityReturnsError() {
        assertThrows(Exception.class, () -> Airport.lookupAirport("Nonexistent City", "XX"));
    }
}

File: kotlin/src/test/kotlin/com/lakamsani/farefinder/AirportTest.kt
package com.lakamsani.farefinder

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFailsWith
import kotlin.test.assertNotNull

class AirportTest {

    @Test
    fun testKnownAirports() {
        val cases = listOf(
            Triple("San Francisco", "CA", "SFO"),
            Triple("New York",      "NY", "JFK"),
            Triple("Chicago",       "IL", "ORD"),
            Triple("Atlanta",       "GA", "ATL"),
            Triple("Denver",        "CO", "DEN"),
            Triple("Seattle",       "WA", "SEA"),
            Triple("Miami",         "FL", "MIA"),
            Triple("Las Vegas",     "NV", "LAS"),
            Triple("Boston",        "MA", "BOS"),
            Triple("Dallas",        "TX", "DFW"),
        )

        for ((city, state, want) in cases) {
            val got = Airport.lookupAirport(city, state)
            assertEquals(want, got, "lookupAirport($city, $state)")
        }
    }

    @Test
    fun testUnknownCityReturnsError() {
        val ex = assertFailsWith<Exception> {
            Airport.lookupAirport("Nonexistent City", "XX")
        }
        assertNotNull(ex.message)
    }
}
